src/ts/common/ui/modal.ts

```diff
@@ -93,31 +97,60 @@ const createTextContainer = (text: string, subTexts?: string[]) => {
 	return container;
 };
 
-const addOnClick = (element: HTMLElement, exit: () => void, onClick?: () => Promise<void>) => {
+const createOnClick = (exit: () => void, onClick?: () => Promise<void>) => {
 	const callback = onClick ?? (() => Promise.resolve());
-	element.addEventListener("click", () => callback().finally(exit));
+	return () => callback().finally(exit);
 };
 
-const dismissModals = (): void => activeOverlays.forEach((modal) => modal.dispatchEvent(new Event("click")));
+const addOnClick = (element: HTMLElement, exit: () => void, onClick?: () => Promise<void>) =>
+	element.addEventListener("click", createOnClick(exit, onClick));
 
-const createModal = ({text, subText, elements, onCancel, colour}: ModalOptions): void => {
-	const exit = () => {
-		activeOverlays.delete(overlay);
-		overlay.remove();
-	};
+const dismissModals = (id = DEFAULT_ID): void =>
```

Comment:
When dismissing modals you should be able to query for the kind of modal it is to prevent you from dismissing unrelated modals